Update authoring support for project references (#919)
microsoft/CsWinRT: docs/authoring.md

## Overview

> **Note**: C#/WinRT authoring and hosting project reference support is available with [.NET 6 RC1](https://dotnet.microsoft.com/download/dotnet/6.0) and later. This means the following scenario is **not supported in .NET 5**: C# app with a project reference to a C#/WinRT Component.

Reviewer: ```suggestion
**Note**: C#/WinRT authoring and hosting project reference support is available with [.NET 6 RC1](https://dotnet.microsoft.com/download/dotnet/6.0) and later. This means the following scenario is **not supported in .NET 5**: C# app with a project reference to a C#/WinRT authored Component.

```

**Note: Authoring Support is still in preview**

> **Update: C++ customers who use project references to CsWinRT components can now use just a single project reference with .NET 6.0 RC1 -- https://dotnet.microsoft.com/download/dotnet/6.0**

Reviewer 2: ```suggestion
**Update: C++ customers can reference CsWinRT components with a single project reference starting in [.NET 6.0 RC1](https://dotnet.microsoft.com/download/dotnet/6.0).

```

>       - **net5.0-windows10.0.17763.0**

Reviewer: Do we want to also list 20348?

> - For project references, you also currently need a reference to the component's generated WinMD. The WinMD can be found in the output directory of the authored component's project. To add both the project and WinMD references, right-click on the native project node, and click **Add** -> **Reference**. Select the C#/WinRT component project under the **Projects** node and the generated WinMD file from the **Browse** node.

Reviewer 2: Maybe instead of the note below, preface this bullet with "If your authored component is built with .NET 5, you also need a reference to the WinMD from referencing C++ apps?

### Consuming from C# applications

> **Note** Starting with C#/WinRT 1.3.5, this requires .NET 6.

Reviewer: Is this true also for package reference?

Author: No, maybe 

`**Note** Starting with C#/WinRT 1.3.5, project references require .NET 6.`
